### PR TITLE
Added cpu topology support for vcpu change method

### PR DESCRIPTION
**Pull request by sathnaga**
This patch enables the support for considering
topology xml change during vcpu changes in xml.

Before this patch xml fails to define incase
of cpu topology present in the xml.

Signed-off-by: Satheesh Rajendran <sathnaga@linux.vnet.ibm.com>
Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

File: virttest/libvirt_xml/vm_xml.py [hunk 1]
```diff
@@ -705,11 +705,27 @@ def set_vm_vcpus(vm_name, value, current=None, virsh_instance=base.virsh):
...
+                # Do change here with respect to vcpu
+                sockets = topology['sockets']
+                cores = topology['cores']
+                threads = topology['threads']
+                if (int(value) % int(threads)) != 0:
+                    # Do not raise test fail here,
+                    # lets leave it to libvirt to detect while define and start
+                    logging.error("vcpu: %s not the multiples of"
+                                  " threads: %s", value, threads)
+                new_cores = int(value)/(int(threads) * int(sockets))
```
Comment (will-Do) [inline virttest/libvirt_xml/vm_xml.py]: This tell us you just update the cores, as you want to make the topology configurable, this is not the good way.

Comment (sathnaga) [inline virttest/libvirt_xml/vm_xml.py]: @will-Do ping

**Comment by balamuruhans** [comment]
LGTM

File: virttest/libvirt_xml/vm_xml.py (hunk 1 above)
Comment (will-Do) [inline virttest/libvirt_xml/vm_xml.py]: I just think we should deal with the calculation/checking in particular cases, and there can set 'sockets', 'cores' and 'threads' directly, so

Comment (sathnaga) [inline virttest/libvirt_xml/vm_xml.py]: @will-Do problem is changing of just vcpu will fail the testcase as it happens in e:g:- vcpucount.py, 
having handled in API will be good as many testcases use this function already.

Comment (sathnaga) [inline virttest/libvirt_xml/vm_xml.py]: @will-Do ping

Comment (will-Do) [inline virttest/libvirt_xml/vm_xml.py]: if the case fail for this reason, then what we need do it fix the case, and keep the API as simple as possible. And there may have some negative cases that we want to set the invalid core/thread or sth else.

Comment (sathnaga) [inline virttest/libvirt_xml/vm_xml.py]: @will-Do Agreed, though the vcpus hotplug going to be core granularity, negative cases will have a probelmand API doing a calculation can also cause a issue later, changed now to take inputs from user.

File: virttest/libvirt_xml/vm_xml.py [hunk 2]
```diff
@@ -693,34 +693,57 @@ def set_pm_suspend(vm_name, mem="yes", disk="yes", virsh_instance=base.virsh):
...
+        :param threads: number of threads, default None
+        :param add_topology: True to add new topology definition if not present
+        :parma virsh_instance: virsh instance
         """
         vmxml = VMXML.new_from_dumpxml(vm_name, virsh_instance=virsh_instance)
-        if value is not None:
+        if vcpus is not None:
             if current is not None:
                 try:
                     current_int = int(current)
```
Comment (balamuruhans) [inline virttest/libvirt_xml/vm_xml.py]: `current_int` is used only once,
```
try:
    if not current:
        try:
            if (int(current) > vcpus):
                raise ...
        except ValueError:
            raise ...
```

Comment (sathnaga) [inline virttest/libvirt_xml/vm_xml.py]: @balamuruhans Tnx, fixed :-)

**Comment by sathnaga** [comment]
@will-Do pls help merge if no further comments, Tnx

**Comment by sathnaga** [comment]
@will-Do Thanks :-)

**Pull request merged**